sdk/typespec/typespec_client_core/src/http/request/mod.rs

```diff
@@ -208,6 +208,35 @@ impl<T> From<RequestContent<T>> for Body {
     }
 }
 
+impl<T> From<Body> for RequestContent<T> {
```

Comment:
So, is the intent here is to have APIs be able to take `RequestContent<T>` and for those APIs be able to take in a `T` to be serialized **or** raw bytes/streams (which I presume would be pre-serialized) to be sent? I'm trying to understand the `T` in this context. The example you show elsewhere takes `ResponseContent<()>`, and I get the case for supporting an arbitrary file stream there, but I'm not clear where we need support for `ResponseContent<T>`.

For the Cosmos DB APIs I've implemented so far, the request body is accepted as just a plain `T` parameter (like `properties: ContainerProperties` when creating a Cosmos DB Container), which gets serialized and added to the body. Of course, that's just one small set of a large Azure SDK surface area, so I'm sure there a number of other use cases I'm not clear on, just trying to understand the purpose of `RequestContent<T>`.